### PR TITLE
Support **kwargs form in `.chunk()`

### LINKED ISSUE #6459: Support **kwargs form in `.chunk()`
### Is your feature request related to a problem?

Take a simple example

```python
import xarray as xr
da = xr.DataArray([1,2,4], dims=['x'])
```
If I want to chunk the array I can do:
```python
da.chunk({'x':1})
```

but I cant do this:
```python
da.chunk(x=1)
```

```
TypeError: chunk() got an unexpected keyword argument 'x'
```

### Describe the solution you'd like

I would like to be able to use the `.chunk()` method for dataarrays in both ways, since it is common for many xarray methods to work either way (e.g. `isel()`/`coarsen()`/etc). 

@TomNicholas 

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by TomNicholas**
Also adds some explicit tests (and type hinting) for `Variable.chunk()`, as I don't think it had dedicated tests before.

- [x] Closes #6459
- [x] Tests added
- [x] User visible changes (including notable bug fixes) are documented in `whats-new.rst`

### CODE REVIEW

File: xarray/core/variable.py [hunk 1]
```diff
@@ -1049,6 +1064,12 @@ def chunk(self, chunks={}, name=None, lock=False):
             )
             chunks = {}
 
+        if isinstance(chunks, (numbers.Number, str, int, tuple, list)):
```
Comment (max-sixty) [inline xarray/core/variable.py:1067]: My understanding is that `Number` is not good for typing: https://github.com/python/mypy/issues/3186, https://stackoverflow.com/questions/50928592/mypy-type-hint-unionfloat-int-is-there-a-number-type

```suggestion
        if isinstance(chunks, (float, str, int, tuple, list)):
```

File: xarray/core/variable.py [hunk 2]
```diff
@@ -1049,6 +1064,12 @@ def chunk(self, chunks={}, name=None, lock=False):
...
+            pass  # dask.array.from_array can handle these directly
+        else:
+            chunks = None if chunks == {} else chunks
```
Comment (max-sixty) [inline xarray/core/variable.py:1070]:
```suggestion
            chunks = chunks or None
```
?

**Review by max-sixty — APPROVED** [review]
Very nice, thanks @TomNicholas !

File: xarray/core/dataarray.py [hunk 3]
```diff
@@ -2,6 +2,7 @@
 
 import datetime
 import warnings
+from numbers import Number
```
Comment (max-sixty) [inline xarray/core/dataarray.py:5]:
```suggestion
```

File: xarray/core/variable.py (hunk 2 above)
Comment (TomNicholas) [reply xarray/core/variable.py:1070]: Actually I think a better fix would be to change line 269 in `either_dict_or_kwargs` 
from
```python
if pos_kwargs is None:
```
to 
```python
if pos_kwargs is None or pos_kwargs == {}:
```

Comment (max-sixty) [reply xarray/core/variable.py]: Yes great idea!

**Comment by TomNicholas** [comment]
Thanks for the rapid review @max-sixty !

**Comment by djhoese** [comment]
This uses the `Number` object but never imports it and is causing CI failures on my unstable dependency environment:

https://github.com/pydata/xarray/blob/ec13944bbd4022614491b6ec479ff2618da14ba8/xarray/core/dataarray.py#L1156

Not sure how the tests didn't hit this or any of the linting checks.

File: xarray/core/dataarray.py [hunk 4]
```diff
@@ -1136,13 +1137,28 @@ def chunk(
...
-        if isinstance(chunks, (tuple, list)):
+        if chunks is None:
+            warnings.warn(
+                "None value for 'chunks' is deprecated. "
+                "It will raise an error in the future. Use instead '{}'",
+                category=FutureWarning,
+            )
+            chunks = {}
+
+        if isinstance(chunks, (Number, str, int)):
```
Comment (djhoese) [inline xarray/core/dataarray.py:1156]: I should have maybe commented here. `Number` is never imported.

**Comment by max-sixty** [comment]
Apologies @djhoese ! Fixing in https://github.com/pydata/xarray/pull/6474.

**Comment by max-sixty** [comment]
I only made suggestions for a subset of the `Number` entries. Probably we should have waited at least for the `pre-commit ci` to pass. 

Something I've done on my other projects is enable "Auto-merging", so you can hit the button and it'll merge when tests pass. Currently GH requires enumerating every check you want to require passing, which then needs to be adjusted every time the test specs change — there's no button for "all". But maybe we should do that.

**Comment by TomNicholas** [comment]
Sorry @djhoese for having an itchy merge finger :sweat_smile:

**Pull request merged**